content/en/docs/guides/build-deploy-software-via-helm-charts-with-ocm-101.md

```diff
@@ -0,0 +1,580 @@
+
+Following the OCM guidelines we will end-up with two component versions and six OCM OCI images.
+
+For building composed components the ["All-in-one"](../getting-started-with-ocm#all-in-one)
+mechanism becomes handy.
+
+### Requirements
+
+- [OCM command line tool](https://github.com/open-component-model/ocm)
+- [kubectl](https://kubernetes.io/docs/reference/kubectl/)
+- [git](https://git-scm.com/downloads)
+- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+- [flux](https://fluxcd.io/flux/installation/#install-the-flux-cli)
+
+## Building OCM Software Artifact
+First we build an OCM OCI image which contains Helm Charts in different kind of formats.
+
+### Prepare Helm Charts
+
+We are leveraging Kubernetes deployments which often use helm charts. The OCM specification supports helm charts as
+an artifact type. For this simple example, we will re-use already existing open source community helm charts.
+
+The OCM CLI supports referencing helm charts being stored in an OCI registry. However most
+publicly available helm charts currently are available from helm chart repositories and
+not from OCI registries. Therefore the helm charts can be embedded in the component archive in three different ways.
+
+Reference a Helm Chart from:
+1. a public Helm Chart repository
+2. local `*.tgz` file
+3. local folder which containing Helm Chart files
+
+First we need to download and unpack the Helm Chart for `podinfo`.
+This can easily be achieved with the helm CLI:
+
+```shell
+helm repo add <repo-name> <helm-chart-repo-url>
+helm pull --destination <target-dir> <repo-name/chart-name>
+```
+
+`podinfo` example:
+```shell
+helm repo add podinfo https://stefanprodan.github.io/podinfo
+helm pull --destination . podinfo/podinfo
+```
+The helm chart is then stored in the current working directory as `podinfo-6.6.0.tgz` and can be referenced as path from there in the `component-constructor.yaml` file (see below).
+
+Unpack `podinfo-6.6.0.tgz` to simulate the process as if this helm chart is our own and is not downloaded from a public repository.
+```shell
+tar -czf podinfo-6.6.0.tgz podinfo
+```
+
+### Input Specification
+
+The corresponding input file for building the component version (`component-constructor.yaml`) will then look like this:
+
+```yaml
+components:
+- name: ${COMPONENT_NAME_PREFIX}/podinfo
+  version: ${PODINFO_VERSION}
+  provider:
+    name: ${PROVIDER}
+  resources:
+  - name: helm-chart-external
+    type: helmChart    
+    relation: external
+    access:
+      type: helm
+      helmChart: podinfo:${PODINFO_CHART_VERSION}
+      helmRepository: https://stefanprodan.github.io/podinfo
+  - name: helm-chart-local-tgz
+    type: helmChart
+    relation: local    
+    input:      
+      type: helm
+      path: podinfo-${PODINFO_CHART_VERSION}.tgz
+  - name: helm-chart-local-folder
+    type: helmChart
+    relation: local    
+    input:   
+      type: dir
+      path: ./podinfo/ 
+  - name: image
```

Comment:
Here there is no distinction between `external` and `local`.
But you could also create it as a local resource:

```yaml
- name: local-image
    type : ociImage
    input:
       type: ociArtifact
       path: ghcr.io/stefanprodan/podinfo:${PODINFO_VERSION} 
       repository: ocm/ocm.software/podinfo/image
```